### DISCUSSION TITLE
Has anyone built Dakota for Apple Silicon (M1 or M2 processor variants) yet?

### DISCUSSION THREAD

**Discussion by esba1ley** [snl-dakota/dakota#76] [Q&A] [answered]
I noticed that the binaries distributed with the release tags on GitHub do not include darwin_arm64 builds.   Is there any intent to do so in the future, or are we left to ourselves to compile the tool suite for ourselves?  If it's the latter, is there a preferred/idiomatic way to do so for the different architecture?

**Comment by jadamstephens** [comment, marked as answer]
We will release Dakota builds for Apple Silicon. I have the machines to do it sitting on my desk. I've had some success using Apple Clang compilers with spack to build the required tools and TPLs.

**Reply by esba1ley** [reply]
Cool!  Thanks!  I’m shepherding a community of new users at JPL, and a lot of us are getting Apple Silicon machines now.  It’s handy to have Dakota local on our laptops for small studies and developing input decks.

**Reply by catherinegorle** [reply]
I wanted to follow up and ask if you have released builds for Apple Silicon. I encourage students in my Uncertainty Quantification class to explore using Dakota for their final projects, but most of us have Apple Silicon machines now.

**Comment by jadamstephens** [comment]
All,

Our Darwin builds will be exclusively arm64 (M1/2/3) beginning with the upcoming May release. We're in the process of transitioning to new build nodes and working out kinks as we find them. I "hand built" a binary package that we've confirmed is portable and works on a couple of macOS versions. Let me know if you have trouble with it:

https://github.com/snl-dakota/dakota/releases/download/v6.19.0/dakota-6.19.0-public-darwin14.Darwin.arm64-cli.tar.gz

**Reply by esba1ley** [reply]
Hi.  I'm curious, which version of MPI should I install on my Apple Silicon Mac to work with this build?  (I'm trying to use MacPorts to do so)

**Reply by esba1ley** [reply]
I've also tried doing it through my conda environment... and I get the same errors:  

```
--------------------------------------------------------------------------
Sorry!  You were supposed to get help about:
    opal_init:startup:internal-failure
But I couldn't open the help file:
    /opt/spack/darwin-sonoma-m1/apple-clang-15.0.0/openmpi-4.1.6-4qhg3r3oysoqnd772pu2r5ras7nhptof/share/openmpi/help-opal-runtime.txt: No such file or directory.  Sorry!
--------------------------------------------------------------------------
--------------------------------------------------------------------------
Sorry!  You were supposed to get help about:
    orte_init:startup:internal-failure
But I couldn't open the help file:
    /opt/spack/darwin-sonoma-m1/apple-clang-15.0.0/openmpi-4.1.6-4qhg3r3oysoqnd772pu2r5ras7nhptof/share/openmpi/help-orte-runtime: No such file or directory.  Sorry!
--------------------------------------------------------------------------
--------------------------------------------------------------------------
Sorry!  You were supposed to get help about:
    mpi_init:startup:internal-failure
But I couldn't open the help file:
    /opt/spack/darwin-sonoma-m1/apple-clang-15.0.0/openmpi-4.1.6-4qhg3r3oysoqnd772pu2r5ras7nhptof/share/openmpi/help-mpi-runtime.txt: No such file or directory.  Sorry!
--------------------------------------------------------------------------
*** An error occurred in MPI_Init
*** on a NULL communicator
*** MPI_ERRORS_ARE_FATAL (processes in this communicator will now abort,
***    and potentially your MPI job)
[MT-510089:04343] Local abort before MPI_INIT completed completed successfully, but am not able to aggregate error messages, and not able to guarantee that all other processes were killed!
```


(the above for N times the number of processor cores I attempted to span) and then... 

```
--------------------------------------------------------------------------
Primary job  terminated normally, but 1 process returned
a non-zero exit code. Per user-direction, the job has been aborted.
--------------------------------------------------------------------------
--------------------------------------------------------------------------
mpirun detected that one or more processes exited with non-zero status, thus causing
the job to be terminated. The first process to do so was:

  Process name: [[1392,1],5]
  Exit code:    1
--------------------------------------------------------------------------
```

**Reply by esba1ley** [reply]
What is the "right" way to install openmpi on Apple Silicon, preferably using MacPorts over homebrew because of their multi-version selection tools.... (that's my package manager of preference for FOSS on Darwin)

**Comment by esba1ley** [comment]
OK.  So I do the following to install and things go (almost) flawlessly.  I do:

1.) download tagged release with a version number from GitHub.com
2.) follow installation instructions
3.) do `sudo xattr -cr <DAKOTA DIR>` which makes the GUI work for versions w/ GUI (see discussion #126 - thanks @jadamstephens !)

I think we can close this thread out.
